doc/devicesinlan_gui.en.1
.TH DEVICESINLAN_GUI 1 2018\-09\-01
.SH NAME

.B devicesinlan_gui:
Scans all devices in your LAN. Then you can set an alias to your known devices in order to detect future strange devices in your net.
.SH SYNOPSIS

devicesinlan_gui [\-\-help] [\-\-version] [\-\-debug DEBUG]
.SH DESCRIPTION

.PP
In the app menu you have the followings features:
.PP
.RS
.B Devices > New Scan
.RE
.PP
.RS
.RS
Searches all devices in tha LAN and show them in a new tab. If some device is not in the known devices list it will be shown with a red background. Devices with a green background are trusted devices
.RE
.RE
.PP
.RS
.B Devices > Show devices database
.RE
.PP
.RS
.RS
Shows all known devices in a new tab.
.RE
.RE
.PP
.RS
.RS
Right click allows you to edit known devices database.
.RE
.RE
.PP
.RS
.B Devices > Load devices list
.RE
.PP
.RS
.RS
Loads a list of known devices in xml format.
.RE
.RE
.PP
.RS
.B Devices > Save devices list
.RE
.PP
.RS
.RS
Saves the known devices list to a xml file.
.RE
.RE
.PP
.RS
.B Devices > Reset database
.RE
.PP
.RS
.RS
Removes all known devices.
.RE
.RE
.PP
.RS
.RS
This option erases all known devices in database.
.RE
.RE
.PP
.RS
.B Configuration > Settings
.RE
.PP
.RS
.RS
In this dialog you can select your prefered language and you can configure the number of concurrence request.
.RE
.RE
.PP
.RS
.B Help > Help
.RE
.PP
.RS
.RS
Shows this help information.
.RE
.RE
.PP
.RS
.B Help > About
.RE
.PP
.RS
.RS
Shows information about DevicesInLAN license and authors.
.RE
.RE
.PP
.RS
.B Help > Check for updates
.RE
.PP
.RS
.RS
Checks for updates in DevicesInLan repository.
.RE
.RE
.PP
.RS
.B Help > Exit
.RE
.PP
.RS
.RS
Exits from program.
.RE
.RE


RELEASES.txt
devicesinlan-0.1.0 <==> svn -r   3 checkout svn://svn.code.sf.net/p/devicesinlan/code/ devicesinlan
devicesinlan-0.2.0 <==> svn -r   5 checkout svn://svn.code.sf.net/p/devicesinlan/code/ devicesinlan
devicesinlan-0.3.0 <==> svn -r   7 checkout svn://svn.code.sf.net/p/devicesinlan/code/ devicesinlan
devicesinlan-0.4.0 <==> svn -r  13 checkout svn://svn.code.sf.net/p/devicesinlan/code/ devicesinlan
devicesinlan-0.5.0 <==> svn -r  15 checkout svn://svn.code.sf.net/p/devicesinlan/code/ devicesinlan
devicesinlan-0.6.0 <==> svn -r  30 checkout svn://svn.code.sf.net/p/devicesinlan/code/ devicesinlan
devicesinlan-0.7.0 <==> svn -r  65 checkout svn://svn.code.sf.net/p/devicesinlan/code/ devicesinlan
devicesinlan-0.8.0 <==> svn -r  69 checkout svn://svn.code.sf.net/p/devicesinlan/code/ devicesinlan
devicesinlan-0.9.0 <==> svn -r  76 checkout svn://svn.code.sf.net/p/devicesinlan/code/ devicesinlan
devicesinlan-0.10.0 <==> svn -r 80 checkout svn://svn.code.sf.net/p/devicesinlan/code/ devicesinlan
devicesinlan-0.11.0 <==> svn -r 85 checkout svn://svn.code.sf.net/p/devicesinlan/code/ devicesinlan
devicesinlan-1.0.0 <==> svn -r 106 checkout svn://svn.code.sf.net/p/devicesinlan/code/ devicesinlan




devicesinlan/devicesinlan_gui.py
#!/usr/bin/python3
import argparse
import sys
import logging
from colorama import init
from PyQt5.QtWidgets import QApplication

#############################

from devicesinlan.libdevicesinlan import Mem, dateversion,  version 
from devicesinlan.ui.frmMain  import frmMain
def main():
    init(autoreset=True)

    app=QApplication(sys.argv)
                
    app.setOrganizationName("DevicesInLAN")
    app.setOrganizationDomain("devicesinlan.sourceforge.net")
    app.setApplicationName("DevicesInLAN")

    parser=argparse.ArgumentParser(prog='devicesinlan_gui', description=app.translate("devicesinlan",'Show devices in a LAN making an ARP search to find them with a user interface'),  
        epilog=app.translate("devicesinlan","If you like this app, please vote for it in Sourceforge (https://sourceforge.net/projects/devicesinlan/reviews/).")+"\n" +app.translate("devicesinlan","Developed by Mariano Muñoz 2015-{}".format(dateversion.year)), 
        formatter_class=argparse.RawTextHelpFormatter
        )
    parser.add_argument('--version', action='version', version="{} ({})".format(version, dateversion))
    parser.add_argument('--debug', help=app.translate("devicesinlan", "Debug program information"))
    args=parser.parse_args()        

    #Por defecto se pone WARNING y mostrar´ia ERROR y CRITICAL
    logFormat = "%(asctime)s %(levelname)s %(module)s:%(lineno)d at %(funcName)s. %(message)s"
    dateFormat='%Y%m%d %I%M%S'

    if args.debug=="DEBUG":#Show detailed information that can help with program diagnosis and troubleshooting. CODE MARKS
        logging.basicConfig(level=logging.DEBUG, format=logFormat, datefmt=dateFormat)
    elif args.debug=="INFO":#Everything is running as expected without any problem. TIME BENCHMARCKS
        logging.basicConfig(level=logging.INFO, format=logFormat, datefmt=dateFormat)
    elif args.debug=="WARNING":#The program continues running, but something unexpected happened, which may lead to some problem down the road. THINGS TO DO
        logging.basicConfig(level=logging.WARNING, format=logFormat, datefmt=dateFormat)
    elif args.debug=="ERROR":#The program fails to perform a certain function due to a bug.  SOMETHING BAD LOGIC
        logging.basicConfig(level=logging.ERROR, format=logFormat, datefmt=dateFormat)
    elif args.debug=="CRITICAL":#The program encounters a serious error and may stop running. ERRORS
        logging.basicConfig(level=logging.CRITICAL, format=logFormat, datefmt=dateFormat)
    else:
        if args.debug:#Bad debug parameter
            logging.basicConfig(level=logging.CRITICAL, format=logFormat, datefmt=dateFormat)
            logging.critical("--debug parameter must be DEBUG, INFO, WARNING, ERROR or CRITICAL")
            sys.exit(1)
        else:     #No debug parameter
            logging.propagate=False

    mem=Mem()
    mem.setApp(app)
    mem.change_language(mem.settings.value("frmSettings/language", "en"))
    mem.setInstallationUUID()

    app.setQuitOnLastWindowClosed(True)
    f = frmMain(mem) 
    f.show()
    sys.exit(app.exec_())

